src/main/proto/inter_node_rpc_service.proto

```diff
@@ -48,40 +48,50 @@ message ResourceContextMessage {
+message AdditionalFields {
+  string name = 1;
+  string description = 2;
 }
 
 extend google.protobuf.EnumValueOptions {
-  ResourceTypeOptions resourceTypeOptions = 50000;
+  AdditionalFields additional_fields = 50000;
 }
 
-enum JvmEnum {
-  OLD_GEN = 0 [(resourceTypeOptions).resourceTypeName = "old gen", (resourceTypeOptions).resourceTypeUnit = "heap usage in percentage"];
-  YOUNG_GEN = 1 [(resourceTypeOptions).resourceTypeName = "young gen", (resourceTypeOptions).resourceTypeUnit = "promotion rate in mb/s"];
-}
+enum ResourceEnum {
+  // JVM
+  OLD_GEN = 0 [(additional_fields).name = "old gen"];
+  YOUNG_GEN = 1 [(additional_fields).name = "young gen"];
+
+  // hardware
+  CPU = 2 [(additional_fields).name = "cpu usage"];
+  IO = 3 [(additional_fields).name = "IO"];
 
-enum HardwareEnum {
-  CPU = 0 [(resourceTypeOptions).resourceTypeName = "cpu usage", (resourceTypeOptions).resourceTypeUnit = "cpu usage in percentage"];
-  IO_TOTAL_THROUGHPUT = 1 [(resourceTypeOptions).resourceTypeName = "IO total throughput", (resourceTypeOptions).resourceTypeUnit = "number of bytes read/written per second"];
-  IO_TOTAL_SYS_CALLRATE = 2 [(resourceTypeOptions).resourceTypeName = "IO total sys callrate", (resourceTypeOptions).resourceTypeUnit = "read and write system calls per second"];
+  // threadpool
+  WRITE_THREADPOOL = 4 [(additional_fields).name = "write threadpool"];
+  SEARCH_THREADPOOL = 5 [(additional_fields).name = "search threadpool"];
 }
 
-enum ThreadPoolEnum {
-  WRITE_QUEUE = 0 [(resourceTypeOptions).resourceTypeName = "write queue", (resourceTypeOptions).resourceTypeUnit = "rejection period in second"];
-  SEARCH_QUEUE = 1 [(resourceTypeOptions).resourceTypeName = "search queue", (resourceTypeOptions).resourceTypeUnit = "rejection period in second"];
+enum MetricEnum {
+  // JVM
+  HEAP_USAGE = 0 [(additional_fields).name = "heap usage", (additional_fields).description = "memory usage in percentage"];
+  PROMOTION_RATE = 1 [(additional_fields).name = "promotion rate", (additional_fields).description = "mb/s"];
+  MINOR_GC = 2 [(additional_fields).name = "minor gc", (additional_fields).description = "time in percentage"];
+
+  // hardware
+  CPU_USAGE = 3 [(additional_fields).name = "cpu usage", (additional_fields).description = "num of cores"];
+  TOTAL_THROUGHPUT = 4 [(additional_fields).name = "total throughput", (additional_fields).description = "number of bytes read/written per second"];
+  TOTAL_SYS_CALLRATE = 5 [(additional_fields).name = "total sys callrate", (additional_fields).description = "read and write system calls per second"];
+
+  // threadpool
+  QUEUE_REJECTION = 6 [(additional_fields).name = "queue rejection", (additional_fields).description = "rejection period in second"];
 }
 
 /*
   message for resource type Enum
 */
-message ResourceType {
-    oneof resource_type_oneof {
-      JvmEnum JVM = 1;
-      HardwareEnum hardware_resource_type = 2;
-      ThreadPoolEnum thread_pool = 3;
-    }
+message Resource {
```

Comment:
We should rename either `message Resource` or the `ResourceEnum resource`. `resourceSummary.getResource().getResource()` isn't very readable.